### PR TITLE
feat: update react-router using vite

### LINKED ISSUE #15203: Migrate off CRA: packages/react/examples/react-router

### PR DESCRIPTION
**Pull request by andreancardona**
Closes #15203 

#### Changelog

**New**

- working version of react-router using vite

**Removed**

- removed old create react app example
- removed unnecessary Tab Navigation example
#### Testing / Reviewing

cd `packages/react/examples/react-router-vite` and run yarn dev

### CODE REVIEW

**Review by tay1orjones — APPROVED** [review]
Looks good! 👍

File: packages/react/examples/react-router-vite/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# `examples/react-router`
+
+> See how to integrate `carbon-components-react` with React Router into a
```
Comment (alisonjoseph) [inline packages/react/examples/react-router-vite/README.md]:
```suggestion
> See how to integrate `@carbon/react` with React Router into a
```

File: packages/react/examples/react-router-vite/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# `examples/react-router`
```
Comment (alisonjoseph) [inline packages/react/examples/react-router-vite/README.md]:
```suggestion
# `examples/react-router-vite`
```

**Review by alisonjoseph — APPROVED** [review]
Couple tiny comments in the readme, but example LGTM!

**Pull request merged**